### PR TITLE
update min number of replicas to 0

### PR DESCRIPTION
**Pull request by sbcd90**
### Description
update min number of replicas to 0.

### Related Issues
Resolves #[Issue number to be closed when this PR is merged]

### Check List
- [X] New functionality includes testing.
- [X] New functionality has been documented.
- [ ] API changes companion pull request [created](https://github.com/opensearch-project/opensearch-api-specification/blob/main/DEVELOPER_GUIDE.md).
- [X] Commits are signed per the DCO using `--signoff`.
- [ ] Public documentation issue/PR [created](https://github.com/opensearch-project/documentation-website/issues/new/choose).

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/security-analytics/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: src/main/java/org/opensearch/securityanalytics/settings/SecurityAnalyticsSettings.java [hunk 1]
```diff
@@ -251,7 +251,7 @@ public static final List<Setting<?>> settings() {
 
     public static final Setting<Boolean> ENABLE_DETECTORS_WITH_DEDICATED_QUERY_INDICES = Setting.boolSetting(
             "plugins.security_analytics.enable_detectors_with_dedicated_query_indices",
-            false,
+            true,
```
Comment (Bukhtawar) [inline src/main/java/org/opensearch/securityanalytics/settings/SecurityAnalyticsSettings.java:254]: Should we keep them as two separate commits?

Comment (sbcd90) [reply src/main/java/org/opensearch/securityanalytics/settings/SecurityAnalyticsSettings.java]: ok reverted this settings change.

**Review by backslasht — APPROVED** [review]

File: src/main/java/org/opensearch/securityanalytics/settings/SecurityAnalyticsSettings.java [hunk 2]
```diff
@@ -12,7 +12,7 @@
 
 public class SecurityAnalyticsSettings {
     public static final String CORRELATION_INDEX = "index.correlation";
-    public static final int minSystemIndexReplicas = 1;
+    public static final int minSystemIndexReplicas = 0;
```
Comment (eirsep) [inline src/main/java/org/opensearch/securityanalytics/settings/SecurityAnalyticsSettings.java:15]: is this for single node cluster scenario?

Comment (sbcd90) [reply src/main/java/org/opensearch/securityanalytics/settings/SecurityAnalyticsSettings.java:15]: yes.

**Review by eirsep — APPROVED** [review]

**Review by AWSHurneyt — APPROVED** [review]

**Review by riysaxen-amzn — APPROVED** [review]
